File: Sample/go/src/tabletest/DataTest.go
package tabletest
import (
    "scorpioproto"
    "container/list"
)

// DataTest 类名
type DataTest struct {
    
    _TestID int32
    _testEnum TestEnum
    _TestDate *list.List
}


// ID   默认值()
func (data *DataTest) ID() int32 { return data._TestID; }

// GetTestID   默认值()
func (data *DataTest) GetTestID() int32 { return data._TestID; }

// GettestEnum   默认值()
func (data *DataTest) GettestEnum() TestEnum { return data._testEnum; }

// GetTestDate   默认值()
func (data *DataTest) GetTestDate() *list.List { return data._TestDate; }


// GetData 获取数据
func (data *DataTest) GetData(key string) interface{} {
    if "TestID" == key {
        return data._TestID;
    }
    if "testEnum" == key {
        return data._testEnum;
    }
    if "TestDate" == key {
        return data._TestDate;
    }
    return nil;
}


// Set 设置数据
func (data *DataTest) Set(value *DataTest) {
    data._TestID = value._TestID;
    data._testEnum = value._testEnum;
    data._TestDate = value._TestDate;
}


// Read 读取数据
func (data *DataTest) Read(fileName string, reader scorpioproto.IScorpioReader) {
    data._TestID = reader.ReadInt32();
    data._testEnum = TestEnum(reader.ReadInt32());
    {
        data._TestDate = list.New();
        number := int(reader.ReadInt32());
        for i := 0; i < number; i++ { 
            data._TestDate.PushBack(Int3Read(fileName, reader)); 
        }
    }
}

// DataTestRead 读取数据
func DataTestRead(fileName string, reader scorpioproto.IScorpioReader) *DataTest {
    ret := new(DataTest)
    ret.Read(fileName, reader)
    return ret
}

File: ScorpioProto/Java/src/Datas/DataTest.java
package Datas;
//本文件为自动生成，请不要手动修改
import java.util.*;
import ScorpioProto.Commons.*;
import ScorpioProto.Table.*;

public class DataTest implements IData {
    
    private Integer _TestID;
    /** 注释  默认值() */
    public Integer getTestID() { return _TestID; }
    public Integer ID() { return _TestID; }
    private TestEnum _testEnum;
    /**   默认值() */
    public TestEnum gettestEnum() { return _testEnum; }
    private List<Int3> _TestDate;
    /**   默认值() */
    public List<Int3> getTestDate() { return _TestDate; }
    private Calendar _TestDateTime;
    /**   默认值() */
    public Calendar getTestDateTime() { return _TestDateTime; }
    private Integer _TestInt;
    /**   默认值(999) */
    public Integer getTestInt() { return _TestInt; }
    
    public Object GetData(String key) {
        if ("TestID".equals(key)) return _TestID;
        if ("testEnum".equals(key)) return _testEnum;
        if ("TestDate".equals(key)) return _TestDate;
        if ("TestDateTime".equals(key)) return _TestDateTime;
        if ("TestInt".equals(key)) return _TestInt;
        return null;
    }
    
    public static DataTest Read(String fileName, IScorpioReader reader) {
        DataTest ret = new DataTest();
        ret._TestID = reader.ReadInt32();
        ret._testEnum = TestEnum.valueOf(reader.ReadInt32());
        {
            List<Int3> list = new ArrayList<Int3>();
            int number = reader.ReadInt32();
            for (int i = 0; i < number; ++i) { list.add(Int3.Read(fileName, reader)); }
            ret._TestDate = Collections.unmodifiableList(list);
        }
        ret._TestDateTime = reader.ReadDateTime();
        ret._TestInt = reader.ReadInt32();
        return ret;
    }
    
    public void Set(DataTest value) {
        this._TestID = value._TestID;
        this._testEnum = value._testEnum;
        this._TestDate = value._TestDate;
        this._TestDateTime = value._TestDateTime;
        this._TestInt = value._TestInt;
    }
    
    @Override
    public String toString() {
        return "TestID:" + _TestID + "," + "testEnum:" + _testEnum + "," + "TestDate:" + _TestDate + "," + "TestDateTime:" + _TestDateTime + "," + "TestInt:" + _TestInt;
    }
}
